algora-io/tv
# (feat) return path to original page

# Return to path not stored when redirecting to login

## Problem
When a user navigates to a protected route (e.g. /ads) while logged out, they get redirected to the login page but after logging in they end up at the home page (/) instead of their original destination.

## Solution
Added `maybe_store_return_to()` to the `redirect_authenticated` function in `RedirectController` to store the return path in the session before redirecting to login.

## Changes
- Modified `lib/algora_web/controllers/redirect_controller.ex` to store return path

## Testing
1. Log out
2. Navigate to /ads
3. Get redirected to login page
4. Log in
5. Verify you get redirected back to /ads

The rest of the return_to flow was already properly implemented in:
- LiveView routes via `redirect_require_login`
- Login links in layouts
- GitHub OAuth flow
- Login callback handling

/claim #101
